frontend/src/features/recommendation/SearchFilter.spec.tsx

```diff
@@ -122,20 +123,35 @@ describe('Filters feature', () => {
     });
   }
 
-  // Click on a language checkbox and verify the resulting URL parameters
-  function clickOnLanguageCheckbox({
-    checkbox,
+  // Select or unselect a language and verify the resulting URL parameters
+  function selectLanguage({
+    language,
+    action = 'add',
     expectInUrl,
   }: {
-    checkbox: 'en' | 'fr' | 'de';
+    language: string;
+    action: 'add' | 'remove';
     expectInUrl: string;
   }) {
-    const languageCheckbox = screen.queryByTestId(
-      'checkbox-choice-' + checkbox
-    );
-    expect(languageCheckbox).not.toBeNull();
-
-    fireEvent.click(languageCheckbox);
+    const languageFilter = queryByTestId(document, 'search-language-filter');
+    const autocomplete = queryByTestId(languageFilter, 'autocomplete');
+    expect(autocomplete).not.toBeNull();
+    const input = within(autocomplete).getByRole('textbox');
+    expect(autocomplete).not.toBeNull();
+
+    if (action === 'add') {
+      autocomplete.focus();
+      userEvent.type(input, language);
```

Comment:
I had to use `userEvent` here because triggering a change with `fireEvent` as suggested in many answers on https://stackoverflow.com/questions/60882089/how-to-test-material-ui-autocomplete-with-react-testing-library didn't work here. I don't know why. It looks like `userEvent` is recommended with mui anyway: https://stackoverflow.com/a/71242000/188760.